### DISCUSSION TITLE
Getting Error : "A programmer is required to upload" while using  megaTinyCore for ATTINY204

### DISCUSSION THREAD

**Discussion by Sourabh0912** [SpenceKonde/megaTinyCore#910] [Q&A]
Hi Spence,
I tried compiling your megaTinyCore for my ATTINY204 IC, but while compiling and uploading code i am getting error **" A programmer is required to upload"**

I also want to read and print ADC data on serial port. Can you please help me.

**Comment by SpenceKonde** [comment]
What programer are you using? Changing fusebytes can only be done via external programmer, and if not using optiboot or if burning bootloader for first time a programmer is needed.

**Reply by Sourabh0912** [reply]
What programer are you using?  : Arduino Uno,

**Reply by Sourabh0912** [reply]
Can you please suggest steps, How to use ATTINY with megaTinyCore.

**Reply by hmeijdam** [reply]
> Can you please suggest steps, How to use ATTINY with megaTinyCore.

Here you go  https://github.com/SpenceKonde/megaTinyCore#getting-started

**Reply by Sourabh0912** [reply]
Thanks for you reply i will try and update.

**Reply by Sourabh0912** [reply]
and can you help me in : Read and print ADC data on serial port using ATTINY204.

**Reply by hmeijdam** [reply]
> and can you help me in : Read and print ADC data on serial port using ATTINY204.

Here you go https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/extras/Ref_Analog.md

**Reply by Sourabh0912** [reply]
ok i will check, Thanks for you reply. :)

**Comment by SpenceKonde** [comment]
Where has this "a programmer is required for upload" come from? Are you using Arduino 2.0.x? I don't think that branch is of release quality yet; it has lots of novel bugs and regressions

**Comment by KWolfe81** [comment]
If using 'Arduino 2.0.x', don't use the arrow 'Upload' button, you must explicitly choose Tools -> Upload using Programmer.

**Comment by SpenceKonde** [comment]
There us almost certainly some incantation I could add to fix this.... but
I don't know what it is.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by SpenceKonde** [reply]
Marked as help wanted and mystery in the hopes that someone who knows how to make normal upgrades use the selected programmer on 2.0.x versions of the IDE will see it and let me in on the secret.
